### PR TITLE
Adds a dependent type meta-annotation

### PR DESCRIPTION
**Pull request by smillst**
Add an annotation to indicate that an annotation element is a Java expression.

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 1]
```diff
@@ -412,6 +408,10 @@ protected String standardizeString(
         }
     }
 
+    public boolean hasDependentAnnotations() {
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: It's obvious, but can you write brief Javadoc anyway?  Thanks!

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 2]
```diff
@@ -433,16 +433,20 @@ private AnnotationMirror standardizeAnnotation(
...
-            }
             AnnotationBuilder builder =
                     new AnnotationBuilder(
                             factory.getProcessingEnv(), AnnotationUtils.annotationName(anno));
-            builder.setValue("value", vpdStrings);
+
+            for (String value : getListOfExpressionElements(anno)) {
+                List<String> expressionStrings =
+                        AnnotationUtils.getElementValueArray(anno, value, String.class, true);
+                List<String> vpdStrings = new ArrayList<>();
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: What does `vpd` stand for in this name?  (Value something?)

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 3]
```diff
@@ -621,12 +625,15 @@ private boolean isExpressionAnno(AnnotationMirror am) {
         }
 
         private List<DependentTypesError> checkForError(AnnotationMirror am) {
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: Can you add documentation for this and other methods you touched?

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 4]
```diff
@@ -725,4 +732,28 @@ protected Boolean reduce(Boolean r1, Boolean r2) {
             }
         }
     }
+
+    private static List<String> getExpressionElementNames(Class<? extends Annotation> clazz) {
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: Please add Javadoc.
That would probably also help me understand why this cannot call `getListOfExpressionElements`.

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 5]
```diff
@@ -725,4 +732,28 @@ protected Boolean reduce(Boolean r1, Boolean r2) {
...
+        for (Method method : methods) {
+            JavaExpression javaExpression = method.getAnnotation(JavaExpression.class);
+            if (javaExpression != null) {
+                elements.add(method.getName());
+            }
+        }
+        return elements;
+    }
+
+    private List<String> getListOfExpressionElements(AnnotationMirror am) {
```
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: Add Javadoc here, too.

File: framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java (hunk 1 above)
Comment (smillst) [inline framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: Whether or not any qualifier in any hierarchy is a dependent annotation.  In other words, does this type system have any dependent annotations.

**Comment by mernst** [comment]
Looks great.  Feel free to merge when the tests pass.

**Pull request merged**